src/classes/frDonorTest.cls

```diff
@@ -89,7 +89,6 @@ public class frDonorTest {
         RestContext.response = res;
         
         createMapping('firstName', 'FirstName');
-        createMapping('lastName', 'LastName');
```

Comment:
Having this mapping was the reason why this test didn't catch it previously.  Because even though we were setting Contact.LastName to null (since in this test, the request is sending null for LastName) we modified the field so to Salesforce we didn't get the exception that we were trying to read the field without querying for it.  Once I removed this line, the test started to fail until I added the changes in frDonor